### ISSUE TITLE
Allow Service.controlWebServer() to silence print

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#31308]
Currently, if you call `Service.controlWebServer(enable: true)`, it prints "Observatory listening on $url" to standard output. This can interfere with the application's output; for example, it breaks the `test` package's reporting, especially for tools using using the JSON reporter. We need a way to control this, such as a `silent: true` parameter.

This is blocking `test`'s ability to support debugging on the Dart VM (dart-lang/test#50).

**Comment by nex3** [comment]
cc @a-siva

**Comment by grouma** [comment]
This is becoming more pressing. @nex3 could we have a temporary workaround that ignores the `Observatory listening..` print?

**Comment by nex3** [comment]
I don't think that's a better investment of engineering effort than fixing the underlying issue.

**Comment by grouma** [comment]
I disagree. We currently support VM debugging in Google3 with Bolt. This support will soon go away and we are left with no alternatives besides Intellij debugging. This seems like an easy fix that can be cleaned up when the underlying issue is resolved.

**Comment by nex3** [comment]
Why *wouldn't* we fix this issue? It's causing user pain (and it will even if IntelliJ hacks around the bogus output), it's a more general solution, and it doesn't seem like a difficult change to make. We're supposed to be a team—hacking around easy fixes in lower-level APIs isn't what a team does.

**Comment by grouma** [comment]
It's not a question of whether we want to fix this or not. We both agree that the underlying issue needs to get resolved. The question is if we can provide support to our customs / users in a time frame that meets their needs. That clearly is not being done given this issue has been open for 23+ days. We either need to make this issue a higher priority, or as I suggested provide a temporary workaround. I'm in favor of doing the latter given that you or I should have the bandwidth.

**Comment by boustrophedon** [comment]
I came across this bug trying to figure out how to debug code run with pub. If you just needed a quick fix @grouma, [this commit](https://github.com/dart-lang/sdk/commit/7e55c12e037ea31e4516813e8b4b0a08de6aa395) seems to have added a `SILENT_OBSERVATORY` env flag that does what you want, but it also does not seem to be working from briefly testing locally. I can confirm that some env variables are being passed to dart because `DART_VM_OPTIONS` is working for me in order to enable the observatory for pub run.

**Comment by a-siva** [comment]
would printing to stderr make it not interfere with the output from test package? That would be a very easy change

**Comment by nex3** [comment]
That would help some, but it would still present a really confusing user experience. What we really want is a way to disable the output entirely.

**Comment by DanTup** [comment]
Is there any progress being made on this? I'd really like to get get VM tests displayed in VS Code the same way that Flutter tests are but it seems like this isn't possible today (without ditching the debugger, which doesn't feel like a good trade).

I'd happily take non-json output in the stdio stream if it meant I could delivery this working - I already have to skip non-json in many of the protocols anyway and I only process lines that are surrounded with `{}` or `[]` (though I appreciate me being happy with it doesn't make it a good idea ;-))

I'd also be happy to run directly through `dart` instead of `pub` if there's a way to get it to use the JSON reporter (as far as I can tell, that's not possible?).

**Comment by devoncarew** [comment]
Given that this has been open for two years, I'm dropping the severity from P1 to P2.

@bkonyi - do you know the level of effort required for this (for having a call to `Service.controlWebServer(enable: true)` not cause an Observatory... message printed to stdout)? If this is something you could get to, do you mind triaging it into a coming beta / stable milestone? https://github.com/dart-lang/sdk/milestones - generally the month milestones mean that we branch at the beginning of that month
